src/ffi/ios/structs.rs

```diff
@@ -0,0 +1,208 @@
+                    .map(|v| unsafe { CFArray::<CFData>::from_void(v) }),
+            )
+        {
+            let key_raw: Cow<_> = (&(*key)).into();
+            let key = HeaderName::try_from(key_raw.as_ref()).context("invalid header name")?;
+
+            for value in values.iter() {
+                let value = HeaderValue::try_from(value.bytes()).context("invalid header value")?;
+                hm.append(&key, value);
+            }
+        }
+
+        Ok(super::Request(ret))
+    }
+}
+
+/// C-serializable HTTP response
+#[repr(C)]
+pub struct Response {
+    /// HTTP code
+    status: u16,
+    /// HTTP headers
+    ///
+    /// Underlying type is CFDictionary<CFString, CFArray<CFData>>,
+    headers: CFDictionaryRef,
+    /// HTTP body
+    body: CFDataRef,
+}
+
+impl TryFrom<super::Response> for Response {
+    type Error = anyhow::Error;
+
+    fn try_from(resp: super::Response) -> anyhow::Result<Self> {
+        let (parts, body) = resp.0.into_parts();
+
+        let mut headers_with_vec = HashMap::with_capacity(parts.headers.keys_len());
+        let mut last_header_name = None;
+        for (k, v) in parts.headers {
+            let k = k
+                .map(|k| k.as_str().to_owned())
+                .or(last_header_name)
+                .context("no header name")?;
+            last_header_name = Some(k.clone());
+
+            headers_with_vec.entry(k).or_insert_with(Vec::new).push(v);
+        }
+
+        let headers = CFDictionary::from_CFType_pairs(
+            &headers_with_vec
+                .into_iter()
+                .map(|(k, vs)| {
+                    (
+                        CFString::new(&k),
+                        CFArray::from_CFTypes(
+                            &vs.into_iter()
+                                .map(|v| CFData::from_buffer(v.as_bytes()))
+                                .collect::<Vec<_>>(),
+                        ),
+                    )
+                })
+                .collect::<Vec<_>>(),
+        );
+
+        let body = CFData::from_buffer(&body);
+
+        let ret = Self {
+            status: parts.status.as_u16(),
+            headers: headers.as_concrete_TypeRef(),
+            body: body.as_concrete_TypeRef(),
+        };
+
+        // ownership for the caller
+        mem::forget(headers);
+        mem::forget(body);
+
+        Ok(ret)
+    }
+}
+
+/// C-serializable `Result<T, String>` value
+#[repr(C)]
+pub union ResultUnion<T> {
+    /// Success value
+    ok: ManuallyDrop<T>,
+    /// String error value
+    err: CFStringRef,
+}
+
+/// C-serializable [`Result`]
+#[repr(C)]
+pub struct Result<T> {
+    /// Is `value.ok` field valid?
+    is_ok: bool,
+    /// Contained value
+    value: ResultUnion<T>,
+}
+
+impl<T> From<anyhow::Result<T>> for Result<T> {
+    fn from(res: anyhow::Result<T>) -> Self {
+        let is_ok = res.is_ok();
+
+        let value = match res {
+            Ok(ok) => ResultUnion {
+                ok: ManuallyDrop::new(ok),
```

Comment:
Where is this dropped?

Reply:
It is not, as it is returned to the caller and is its responsability to clean it when done with it.

Reply:
And does that actually happen?

Reply:
By the magic of refcounting and the GC of swift, it does.

If you look at `ios/lightarti-rest.h` (after `make`), you can see the resulting function. Every `Result` only contain values or pointer in the CoreFoundation land, which are taken care of by swift.